src/schemas/execute.yaml

```diff
@@ -0,0 +1,121 @@
+    stateOverrides:
+      title: State overrides
+      $ref: '#/components/schemas/StateOverrides'
+    calls:
+      title: List of calls
+      $ref: '#/components/schemas/Calls'
+Calls:
+  title: List of message calls
+  type: array
+  items:
+    $ref: '#/components/schemas/Call'
+Call:
+  title: Call + block overrides
+  type: object
+  required:
+    - transactionArgs
+  properties:
+    transactionArgs:
+      title: Call arguments
+      $ref: '#/components/schemas/GenericTransaction'
+    blockOverrides:
+      title: Overrides for block metadata
+      $ref: '#/components/schemas/BlockOverrides'
+StateOverrides:
+  title: Accounts in state to be overridden
+  type: object
+  additionalProperties:
+    $ref: '#/components/schemas/AccountOverride'
+AccountOverride:
+  title: Details of an account to be overridden
+  type: object
+  oneOf:
+    - $ref: '#/components/schemas/AccountOverrideState'
+    - $ref: '#/components/schemas/AccountOverrideStateDiff'
+AccountOverrideState:
+  title: Account override with whole storage replacement
+  properties:
+    nonce:
+      title: Nonce
+      $ref: '#/components/schemas/uint64'
+    balance:
+      title: Balance
+      $ref: '#/components/schemas/uint256'
+    code:
+      title: Code
+      $ref: '#/components/schemas/bytes'
+    state:
+      title: Storage
+      $ref: '#/components/schemas/AccountStorage'
+AccountOverrideStateDiff:
+  title: Account override with partial storage modification
+  properties:
+    nonce:
+      title: Nonce
+      $ref: '#/components/schemas/uint64'
+    balance:
+      title: Balance
+      $ref: '#/components/schemas/uint256'
+    code:
+      title: Code
+      $ref: '#/components/schemas/bytes'
+    stateDiff:
+      title: Storage difference
+      $ref: '#/components/schemas/AccountStorage'
+AccountStorage:
+  title: Storage slots for an account
+  type: object
+  additionalProperties:
+    - $ref: '#/components/schemas/hash32'
+BlockOverrides:
+  title: Context fields related to the block being executed
+  type: object
+  properties:
+    number:
+      title: Number
+      $ref: '#/components/schemas/uint256'
+    difficulty:
+      title: Difficulty
+      $ref: '#/components/schemas/uint256'
+    time:
+      title: Time
+      $ref: '#/components/schemas/uint256'
+    gasLimit:
+      title: Gas limit
+      $ref: '#/components/schemas/uint64'
+    coinbase:
+      title: Coinbase
+      $ref: '#/components/schemas/address'
+    random:
+      title: Random
+      $ref: '#/components/schemas/hash32'
+    baseFee:
+      title: Base fee
+      $ref: '#/components/schemas/uint256'
+CallResults:
+  title: Results of batch call
+  type: array
+  items:
+    $ref: '#/components/schemas/CallResult'
+CallResult:
+  title: Result of a call
+  type: object
+  required:
+    - return
+  properties:
+    return:
+      title: Return data
+      $ref: '#/components/schemas/bytes'
+    error:
+      title: Execution error
+      type: string
```

Comment:
The result for each transaction should include logs and gas spent as well.  I suspect that all clients are generating this information anyway, and throwing it away is a waste when we could just return it all.  In many cases, this will also make it so the caller doesn't need to do multiple round trips (e.g., once for gas estimate and again for speculative execution and again to read a deterministic event like contract creation details).

If the call is a contract creation (`to: null`), it should also return the address of the created contract for the same reason.

Reply:
> include logs and gas spent as well

@s1na mentioned an argument against including gas spent in the return interface ([here](https://github.com/ethereum/go-ethereum/pull/25743#discussion_r981361476)). The intention I mentioned was to use the gas spent as is for gas limits, which I agree would fail when tx. However, including gas spent in the response could still be useful if utilized with care (passing a factor of more gas for e.g. to account for 63/64 gas passed to child msg calls). And in the documentation, a note can be mentioned that gas spent is not accurate to be used as gas limit.

Reply:
I personally don't think it is worth trying to optimize the edge case where someone does `if (GAS) ...` because that is very rare and when it is used, the dapp almost always explicitly sets the GAS required anyway so the returned gas used won't matter.

I think the value add of returning gas used far outweighs the downsides of it being wrong in a handful of esoteric scenarios.  I *do* recommend that we standardize on the amount of gas provided to each call in `eth_multicall` though.  Perhaps `block.gas_limit - gas_used_in_previous_transactions_in_multicall`?  Essentially, assume that the whole set of transactions is intended to fit inside a block (this also provides a sanity check on total gas used) and they "burn through" the block's gas as they are executed.

This would prevent people from using this to do mega gas operations, but I think we can address that by simply having `eth_multicall` accept a `gas_limit` parameter for the `block` and each `transaction` so this isn't a hard limiter.

Reply:
I still think `gasUsed` should not be used as an estimate. But I can sympathise with adding this info to the call. It comes at no overhead. Right now all this can be fetched in one call through geth's tracing API (via the `callTracer`), but I agree it should be part of `eth_`. I'm happy to add these.

Re `gas_limit` there is a need for a block gas limit set in the client itself. Because providers like Infura will want to cap the runtime of API methods. Geth has both a gas limit and a deadline in seconds.

Reply:
Setting an upper bound on the gas able to be used by an `eth_call` or `eth_multicall` feels like it is out of scope of this schema.  On my local node, I should be able to do calls that use trillions of gas, and the fact that a centralized service (which we should not be catering our designs toward) needs rate limiting should not negatively impact a self-hosted operator from doing whatever they want.

Reply:
To clarify: what I meant was there is a CLI flag anyway that will set a limit for all `eth_call`s and `eth_multicall`s. On your local machine you're free to set it to trillions. Question is whether a per-invocation limit is also useful.

> This would prevent people from using this to do mega gas operations,

Adding a `gasLimit` parameter won't solve this for public facing nodes. They will simply set a high gas limit.

Reply:
> To clarify: what I meant was there is a CLI flag anyway that will set a limit for all `eth_call`s and `eth_multicall`s. On your local machine you're free to set it to trillions. Question is whether a per-invocation limit is also useful.

Such a setting feels like it is out of scope of this standard?  I might be able to be convinced that it is relevant, but rate limiting features and DoS protection generally aren't standardized.

What we need to standardize is "how much gas is given when the parameter is missing" and picking something "reasonable" (e.g., not 2^256, and not 30,000) as a default feels like the right way to go to me.  If a particular client has limits that are lower than that (such as from configuration) then they should return an appropriate rate limiting error and require the user to explicitly set the gas limits in their requests I think if the user exceeds those limits.

Reply:
I think that the `if (GAS) ` case is not really worth considering but the 63/64 and the refunds are. 

However, there's really two uses for gas that should be handled:
1.) what you need to add to transaction for it to succeed (eth_estimateGas/eth_batchEstimateGas)
2.) What you actually pay for, some way to determine what my EOA is actually going to be charged for the transaction. It feels most natural for that to occur in the eth_batchCall . If there is a back-run, it is very important to the sender how much gas is going to actually be charged as mev profit can often be extremely small and the difference between the actual gas used and the gas estimation is the difference between profit and loss.